### PR TITLE
add csv& json

### PR DESCRIPTION
**Pull request by XuhuiZhou**
Closes # 

## 📑 Description

Adding helper function to serialize a list of episodes to csv or json format

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: sotopia/database/utils.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    """
+    agents = [AgentProfile.get(pk=agent) for agent in episode.agents]
+
+    return {
+        f"{profile.first_name} {profile.last_name}": f"{profile.first_name} {profile.last_name} is a {profile.age}-year-old {_map_gender_to_adj(profile.gender)} {profile.occupation.lower()}. {profile.gender_pronoun} pronouns. {profile.public_info} Personality and values description: {profile.personality_and_values} {profile.first_name}'s secrets: {profile.secret}"
+        for profile in agents
+    }
+
+
+def get_social_goals_from_episode(
```
[the remark this answers is not in the document]
Comment (ProKil) [reply sotopia/database/utils.py:71]: The output format should also be specified, e.g. `get_agent_name_to_social_goal_from_episodes`

File: sotopia/database/utils.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        "scenario": [
+            get_scenario_from_episode(episode) for episode in episodes
+        ],
+        "codename": [
+            get_codename_from_episode(episode) for episode in episodes
+        ],
+        "agents_background": [
+            get_agents_background_from_episode(episode) for episode in episodes
+        ],
+        "social_goals": get_social_goals_from_episode(episodes),
```
Comment (ProKil) [inline sotopia/database/utils.py:150]: why these two are not following the same as previous ones? like key: [get...(episode) for episode in episodes]

**Review by ProKil — CHANGES_REQUESTED** [review]

File: sotopia/database/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (XuhuiZhou) [reply sotopia/database/utils.py:71]: done!

File: sotopia/database/utils.py (hunk 2 above)
Comment (XuhuiZhou) [reply sotopia/database/utils.py:150]: fixed!

File: sotopia/database/utils.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        "social_interactions": get_social_interactions_from_episode(episodes),
+    }
+    df = pd.DataFrame(data)
+    df.to_csv(filepath, index=False)
+
+
+def episodes_to_json(
```
[the remark this answers is not in the document]
Comment (XuhuiZhou) [reply sotopia/database/utils.py:157]: clear!

File: notebooks/redis_stats.ipynb
Comment (ProKil) [inline notebooks/redis_stats.ipynb:1]: it seems that line 5 is not compatible with the current main b/c the depreciation of `LLM_Name`. Could you git merge main and check again?

File: sotopia/database/utils.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+import pandas as pd
+from pydantic import ConstrainedList, conlist, root_validator
+from redis_om import HashModel, JsonModel
+from redis_om.model.model import Field
+
+from .logs import EpisodeLog
+from .persistent_profile import AgentProfile, EnvironmentProfile
+
+
+class TwoAgentEpisodeWithScenarioBackgroundGoals(JsonModel):
```
Comment (ProKil) [inline sotopia/database/utils.py:12]: Does it need to be `redis_om.JsonModel`? Or just `pydantic.BaseModel`. I would also suggest removing `index=True` and make these non-optional.

File: tests/database/test_database.py
Comment (ProKil) [inline tests/database/test_database.py:1]: This is a question not a request change: Can you describe what did you change here? I don't really understand. It seems that the functions that you added are not the new test here.

**Review by ProKil — CHANGES_REQUESTED** [review]

File: tests/database/test_database.py
Comment (XuhuiZhou) [reply tests/database/test_database.py:1]: It's just a better way to create dummy episode to test things shared by many other functions

File: sotopia/database/utils.py (hunk 4 above)
Comment (XuhuiZhou) [reply sotopia/database/utils.py:12]: Oh why? I think all of our classes use jsonModel?

Comment (ProKil) [reply sotopia/database/utils.py:12]: JsonModel inherits pydantic.BaseModel with extra functions of interaction with redis. For example, "index=True" doesn't have any meaning here.

File: sotopia/database/utils.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        for episode in episodes:
+            data = TwoAgentEpisodeWithScenarioBackgroundGoals(
+                episode_id=episode.pk,
+                scenario=get_scenario_from_episode(episode),
+                codename=get_codename_from_episode(episode),
+                agents_background=get_agents_background_from_episode(episode),
+                social_goals=get_agent_name_to_social_goal_from_episode(
+                    episode
+                ),
+                social_interactions=get_social_interactions_from_episode(
```
Comment (ProKil) [inline sotopia/database/utils.py:173]: Add GPT-4 rewards?

**Review by ProKil — CHANGES_REQUESTED** [review]

**Review by ProKil — APPROVED** [review]

**Pull request merged**